code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm

```diff
@@ -463,5 +468,35 @@
 	piping_layer = 4
 	icon_state = "scrub_map_on-4"
 
+/obj/machinery/atmospherics/components/unary/vent_scrubber/plunger_act(obj/item/plunger/P, mob/living/user, reinforced)
+	if(!clogged)
+		return
+
+	if(welded)
+		to_chat(user, span_notice("You cannot pump the [name] if it's welded shut!."))
+		return
+
+	to_chat(user, span_notice("You begin pumping the [name] with your plunger."))
+	if(do_after(user, 60, target = src))
+		to_chat(user, span_notice("You finish pumping the [name]."))
+		clogged = FALSE
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/clog(datum/source)
+	SIGNAL_HANDLER
+	RegisterSignal(src, COMSIG_PRODUCE_MOB, .proc/produce_mob)
```

Comment:
Same here. Instead of being a signal, you can just call "produce_mob" directly, instead of controlling it via signal.